connectors/sources/mongo.py

```diff
@@ -256,24 +257,51 @@ async def validate_config(self):
         await super().validate_config()
 
         client = self.client
+
         configured_database_name = self.configuration["database"]
         configured_collection_name = self.configuration["collection"]
 
-        existing_database_names = await client.list_database_names()
+        # First check if collection is accessible
+        collection_valid = True
+
+        try:
+            # This works on both standalone and Managed mongo in the same way
+            await client[configured_database_name].validate_collection(
+                configured_collection_name
+            )
+        except OperationFailure:
+            collection_valid = False
+
+        if collection_valid:
+            return
```

Comment:
why not return as a last step in the above `try` block?

Reply:
Omg yes, will do it

Reply:
Addressed in https://github.com/elastic/connectors/pull/2023/commits/4c51560fa6c1d8de997cea0b9d580207cc3cd806